CMakeLists.txt

```diff
@@ -92,12 +114,48 @@ endif()
 target_compile_definitions(ccls PRIVATE
                            DEFAULT_RESOURCE_DIRECTORY=R"\(${Clang_RESOURCE_DIR}\)")
 
+
+### use local installed RapidJSON or FETCH from git via cmake
+find_package(RapidJSON QUIET)
+if(RapidJSON_FOUND)
+  message("${BoldBlue}\nRapidJSON has been installed by your system before cloning ccls\n\n${ColourReset}")
+  include_directories(${RapidJSON_INCLUDE_DIRS})
+else()
+  message("${Green}====> Downloading RapidJSON ---- A fast JSON parser/generator for C++ with both SAX/DOM style API http://rapidjson.org/
```

Comment:
I am not sure if FetchContent_Declare is better than `git submodule update --init` or `git clone --recursive`: it looks more opaque and the term "downloading" would actually worry me: in general I don't like the build system downloads something without my attention.

Reply:
> it looks more opaque and the term "downloading" would actually worry me: in general I don't like the build system downloads something without my attention.

fair enough. ccls uses only one git submodule, so `git submodule update update --init` is very close to `FetchContent`. So the only pros left is to use local rapidjson when possible. Generally speaking, FetchContent and git submodule are both fine for me. Feel free to close this pr if you don't like FetchContent. 

As for the term "downloading", I cannot think of better alternatives haha.